test/integration-new/cni/service_connectivity_test.go

```diff
@@ -0,0 +1,177 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cni
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/manifest"
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/utils"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	appsV1 "k8s.io/api/apps/v1"
+	batchV1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	serviceLabelSelectorKey = "role"
+	serviceLabelSelectorVal = "service-test"
+)
+
+// Verifies connectivity to deployment behind different service types
+var _ = Describe("test service connectivity", func() {
+	var err error
+
+	// Deployment running the http server
+	var deployment *appsV1.Deployment
+	var deploymentContainer v1.Container
+
+	// Service front ending the http server deployment
+	var service *v1.Service
+	var serviceType v1.ServiceType
+	var serviceAnnotation map[string]string
+
+	// Test job that verifies connectivity to the http server
+	// by querying the service URL
+	var testerJob *batchV1.Job
+	var testerContainer v1.Container
+
+	// Test job that verifies test fails when connecting to
+	// a non reachable port/address
+	var negativeTesterJob *batchV1.Job
+	var negativeTesterContainer v1.Container
+
+	JustBeforeEach(func() {
+		deploymentContainer = manifest.NewBusyBoxContainerBuilder().
+			Image("python").
+			Command([]string{"python3"}).
+			Args([]string{"-m", "http.server", "80"}).Build()
```

Comment:
What about HTTPS? Are you planning to add it later?

Reply:
Good point! I have taken up an AI to add HTTPS tests later.